### PR TITLE
Add some tests for dialyzer warnings and silence remote call warnings

### PR DESCRIPTION
**Pull request by fishcakez**
The raise test fails for two reasons:
* Generated `exception/1` specs are too complex
* Missing feature to mark quoted AST as generated

We should report/fix the issue upstream in the TODO comment in elixir_translator.erl about generated matches.

### CODE REVIEW

File: lib/elixir/test/elixir/dialyzer_test.exs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      |> String.to_char_list()
+    File.cp!(base_plt, plt)
+
+    dialyzer = [analysis_type: :succ_typings, check_plt: false,
+      files_rec: [dir], plts: [plt]]
+
+    {:ok, [outdir: dir, dialyzer: dialyzer]}
+  end
+
+  test "No warnings on valid remote calls", context do
```
Comment (lexmag) [inline lib/elixir/test/elixir/dialyzer_test.exs]: Let's not set a precedent of capitalized test names. :bowtie:

Comment (fishcakez) [inline lib/elixir/test/elixir/dialyzer_test.exs]: Thanks @lexmag, bit rusty after 2 weeks away

**Comment by fishcakez** [comment]
> Generated exception/1 specs are too complex

Fixed this.

**Comment by fishcakez** [comment]
In master.

**Comment by fishcakez** [comment]
Missing raise tests in master

**Pull request closed without merging**